code/modules/projectiles/guns/energy.dm

```diff
@@ -129,10 +128,13 @@
 	return
 
 /obj/item/gun/energy/update_icon()
-	..()

```

Comment:
There's nothing stopping energy guns having bayonets/flashlights is there ?

Reply:
Correct but that's now handled in the attachment proc for guns